### PR TITLE
Fix index route

### LINKED ISSUE #43: Root path / opening comment
From [the opening comments](https://github.com/fortunejs/fortune-json-api/blob/5ef7ad19604755cba1a598b3d9dbada1394c2b8a/lib/index.js#L32-L35):

> For example, it assumes that the client has *a priori* knowledge of types that exist on the server, since it does not define an entry point.

This is incorrect, I think. From @steveklabnik, one of the authors of JSON API:

> The root path looks like any other JSON API response, we don't need a special one.

(Source: https://github.com/json-api/json-api/issues/126#issuecomment-37676482 )

This leaves a bit open for interpretation, but the gist is that you can just return some top-level `links`, and any other information that you would like. Here's one simple root endpoint that I've generated (this is returned as the root of `/v1`. For the root before that, you may link to the different hosted versions rather than to the resources, or if there's only one version available, you may just redirect to the root of that):

```json
{
  "jsonapi": {
    "version": "1.0",
    "meta": {
      "extensions": []
    }
  },
  "meta": {
    "api_version": "1"
  },
  "links": {
    "cats": {
      "href": "/v1/cats",
      "meta": {
        "supported_actions": [
          "create",
          "read_one",
          "read_many",
          "update",
          "delete"
        ]
      }
    },
    "categories": {
      "href": "/v1/categories",
      "meta": {
        "supported_actions": [
          "create",
          "read_one",
          "read_many",
          "update",
          "delete"
        ]
      }
    },
    "people": {
      "href": "/v1/people",
      "meta": {
        "supported_actions": [
          "create",
          "read_one",
          "read_many",
          "update",
          "delete"
        ]
      }
    }
  }
}
```

The comment makes the point that you need a priori knowledge of the resources, but one of my favorite features of JSON API is its commitment to hypermedia. You should only need a priori knowledge of the root path of the API + the JSON API specification itself to use it. From there, the rest is given to you (you can imagine adding more `meta` for each resource that informs you of the relationships the resource has, the attributes it has, and so on, to enable you to use the different features of JSON API, like `includes`, `fields`, etc.).

**Update:** I just spun up an instance of this, and I see that the root endpoint is returned, but it's not a valid JSON API response. Would you be open to a PR that fixes that?

---

Another line in the opening comment is:

> From a web architecture standpoint, it assumes tight coupling with the API consumer and the HTTP protocol.

This is true, for now, and it's especially easy for one to come away with this conclusion, given that the documentation is heavily centered around transporting the data over HTTP. But JSON API *is* being developed with the goal of one day being protocol agnostic. This is one reason why it is recommended that error codes be returned in the error object. I can find the source on that in the issues if you're interested, too!

--

Last, this opening comment communicates a lot of opinions about JSON API, which is a really good topic for a blog post, but I'm not sure if it's fitting for this library. For instance:

> and relationship entities which are an entirely unnecessary complication.

You may be correct here. But is this the right place to make that argument? Do what you think is best, o' course, but I think it'd be more helpful if the comments here explained the _code_, rather than your opinions on JSON API.

In any event, I am pretty interested to hear more on your thoughts of JSON API. I get the sense that it isn't positive, but I'd love to hear more if you do have a blog 😛

### PR DESCRIPTION
**Pull request by jamesplease**
In #45 , there are 13 failing tests. This fixes 2 of them. 11 to go!

---

I separated out the code changes from the comment changes. Feel free to drop the comment changes / make adjustments as you see fit, @daliwali . I based the changes on our conversation over in #43 

Also, I originally thought the string links were making the responses invalid, but I was wrong! String links are valid. It was the absence of one of: `data`, `errors`, or `meta` that was invalid. My mistake.

---

Resolves #43

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -138,13 +138,15 @@ module.exports = Serializer => {
 
       contextResponse.payload = {
         [reservedKeys.jsonapi]: jsonapi,
+        [reservedKeys.meta]: {},
```
Comment (jamesplease) [inline lib/index.js:141]: Might seem weird, but all responses **must** contain either data, errors, or meta. We definitely have no data nor errors at the root, so I think it makes the most sense to include an empty `meta`.

[Source](http://jsonapi.org/format/#document-top-level)

File: lib/index.js [hunk 2]
```diff
@@ -138,13 +138,15 @@ module.exports = Serializer => {
...
         [reservedKeys.links]: {}
       }
 
       for (let type in recordTypes) {
         if (inflectType) type = inflection.transform(type, typeInflections[1])
-        contextResponse.payload[reservedKeys.links][type] = prefix +
-          uriTemplate.fillFromObject({ type })
+        contextResponse.payload[reservedKeys.links][type] = {
+          href: prefix + uriTemplate.fillFromObject({ type })
+        }
```
Comment (jamesplease) [inline lib/index.js]: The "links object" definition can be seen [here](http://jsonapi.org/format/#document-links).

**Comment by jamesplease** [comment]
Let me know if there's anything else you'd like to see here before this gets merged, @daliwali . It looks good on my side of things.

**Comment by gr0uch** [comment]
Main purpose of this seems to be getting schema validators passing, it will likely have no effect on real clients. I'm fine with this since it has effectively no impact.

**Pull request merged**